### PR TITLE
[Fix](Planner) add date/month/year add function fe constant fold with dateV2/dateTimeV2

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
## Proposed changes

Problem and Cause:
In original planner, date_add function would choose different data type of datetime or datev2. Which when original planner choose datev2 as constant date type. Fe could not folding date_add function because missing of function signature of folding constant date_add(datev2, int)

Solved:
Add corresponding function signatures of date_add/months_add/years_add in original planner

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java [hunk 1]
```diff
@@ -115,18 +125,54 @@ public static DateLiteral yearsAdd(LiteralExpr date, LiteralExpr year) throws An
...
     @FEFunction(name = "days_add", argTypes = { "DATETIME", "INT" }, returnType = "DATETIME")
     public static DateLiteral daysAdd(LiteralExpr date, LiteralExpr day) throws AnalysisException {
         DateLiteral dateLiteral = (DateLiteral) date;
         return dateLiteral.plusDays(day.getLongValue());
     }
 
+    @FEFunction(name = "days_add", argTypes = { "DATEV2", "INT" }, returnType = "DATEV2")
+    public static DateLiteral daysAddDateV2(LiteralExpr date, LiteralExpr day) throws AnalysisException {
+        DateLiteral dateLiteral = (DateLiteral) date;
+        return dateLiteral.plusYears(day.getLongValue());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java:167]: plusDays?

File: fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java [hunk 2]
```diff
@@ -115,18 +125,54 @@ public static DateLiteral yearsAdd(LiteralExpr date, LiteralExpr year) throws An
...
+    @FEFunction(name = "months_add", argTypes = { "DATEV2", "INT" }, returnType = "DATEV2")
+    public static DateLiteral monthsAddDateV2(LiteralExpr date, LiteralExpr month) throws AnalysisException {
+        DateLiteral dateLiteral = (DateLiteral) date;
+        return dateLiteral.plusYears(month.getLongValue());
+    }
+
+    @FEFunction(name = "months_add", argTypes = { "DATETIMEV2", "INT" }, returnType = "DATETIMEV2")
+    public static DateLiteral monthsAddDateTimeV2(LiteralExpr date, LiteralExpr month) throws AnalysisException {
+        DateLiteral dateLiteral = (DateLiteral) date;
+        return dateLiteral.plusYears(month.getLongValue());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java:155]: plusMonths?

**Review by morrySnow — CHANGES_REQUESTED** [review]

File: regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy [hunk 3]
```diff
@@ -222,6 +222,10 @@ suite("test_date_function") {
     qt_sql """ select date_add(test_datetime, INTERVAL 2 MINUTE) result from ${tableName}; """
     qt_sql """ select date_add(test_datetime, INTERVAL 2 SECOND) result from ${tableName}; """
 
+    String explainResult
+    explainResult = sql("select * from ${tableName} where test_datetime >= date_add('2024-01-16',INTERVAL 1 day);")
```
Comment (gavinchou) [inline regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy:226]: add more functions

File: fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java (hunk 2 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java (hunk 1 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/rewrite/FEFunctions.java]: done

File: regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy (hunk 3 above)
Comment (LiBinfeng-01) [reply regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy:226]: done

**Comment by LiBinfeng-01** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy [hunk 4]
```diff
@@ -222,6 +222,14 @@ suite("test_date_function") {
...
+    assertFalse(explainResult.contains("date_add"))
+    explainResult = sql("select * from ${tableName} where test_datetime >= months_add('2024-01-16',1);")
+    assertFalse(explainResult.contains("months_add"))
+    explainResult = sql("select * from ${tableName} where test_datetime >= years_add('2024-01-16',1);")
+    assertFalse(explainResult.contains("years_add"))
```
Comment (morrySnow) [inline regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy:231]: we should check the result is right, wo need to check contains '2025-01-16'. btw, u should use
```
explain {
    sql """ ... """
    contains "...."
}
```

Comment (LiBinfeng-01) [reply regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_function.groovy]: done

**Review by morrySnow — DISMISSED** [review]

**Review by gavinchou — APPROVED** [review]
LGTM

**Comment by gavinchou** [comment]
2.0 #34484

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run p0

**Review by starocean999 — APPROVED** [review]

**Pull request merged**